ta/pkcs11/src/pkcs11_helpers.c

```diff
@@ -218,4 +470,87 @@ const char *id2str_session_state(uint32_t id)
 {
 	return ID2STR(id, string_session_state, "PKCS11_CKS_");
 }
+
+const char *id2str_attr(uint32_t id)
+{
+	size_t n = 0;
+
+	for (n = 0; n < ARRAY_SIZE(attr_ids); n++) {
+		if (id != attr_ids[n].id)
+			continue;
+
+		/* Skip PKCS11_ prefix */
+		return (char *)attr_ids[n].string + strlen("PKCS11_CKA_");
```

Comment:
cast not needed,`.string` is already a `const char *`.

Maybe shrink to:
```c
	for (n = 0; n < ARRAY_SIZE(attr_ids); n++)
		if (id == attr_ids[n].id)
			return attr_ids[n].string + strlen("PKCS11_CKA_");

Reply:
I'll remove the casting, but I'll keep the comment.